qiskit/transpiler/passes/synthesis/solovay_kitaev_synthesis.py

```diff
@@ -139,21 +136,31 @@ def __init__(
         self,
         recursion_degree: int = 3,
         basic_approximations: str | dict[str, np.ndarray] | None = None,
+        *,
+        basis_gates: list[str | Gate] | None = None,
+        depth: int = 16,
     ) -> None:
-        """
+        r"""
         Args:
             recursion_degree: The recursion depth for the Solovay-Kitaev algorithm.
                 A larger recursion depth increases the accuracy and length of the
                 decomposition.
             basic_approximations: The basic approximations for the finding the best discrete
-                decomposition at the root of the recursion. If a string, it specifies the ``.npy``
+                decomposition at the root of the recursion. If a string, it specifies the
                 file to load the approximations from. If a dictionary, it contains
-                ``{label: SO(3)-matrix}`` pairs. If None, a default based on the H, T and Tdg gates
-                up to combinations of depth 10 is generated.
+                ``{label: SO(3)-matrix}`` pairs. If ``None``, a default based on the :math:`H`,
+                 :math:`T` and :math:`T^\dagger` gates up to depth 16 is generated.
+            basis_gates: The basis gates used to build the net of basic approximations.
+                Defaults to ``["h", "t", "tdg"]``. This argument is incompatible with
+                ``basic_approximations``.
+            depth: The maximal gate depth used in basic approximations. This argument is
+                incompatible with ``basic_approximations``.
```

Comment:
please explain this "incompatible". what should the user provide - either basic_approximations or basis_gates and depth?

Reply:
is there an error if the user provides both?

Reply:
exactly, I'll reformulate a bit